### PR TITLE
Fix propositions return value

### PR DESCRIPTION
**Pull request by jonsnyder**
## Description

While writing documentation wiki pages for the apply propositions, I noticed that the send event call was not returning anything in propositions. It was always an empty array. After investigation, I found that this was caused by the DecisioningEngine also returning propositions from its onBeforeEvent lifecycle method. This PR changes how the lifecycle return values are merged together. If the merged values are arrays, then the arrays are concat'ed together instead of overwritten.

Additionally, I found a few more tests failing because experience edge is now returning default-content-items when before it wasn't. I've updated the tests to be more lenient on their expectations regarding the number of propositions returned.

Lastly, there are two tests for applyResponse that I skipped because I noticed there are a bunch of new propositions returned from experience edge that are messing up the expectations. See https://jira.corp.adobe.com/browse/PDCL-11302

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [ ] I have made any necessary test changes and all tests pass.
- [ ] I have run the Sandbox successfully.

### CODE REVIEW

File: test/unit/specs/core/edgeNetwork/mergeLifecycleResponses.spec.js [hunk 1]
```diff
@@ -70,6 +70,9 @@ describe("mergeLifecycleResponses", () => {
                 ]
               }
             ]
+          },
+          {
+            propositions: []
```
Comment (carterworks) [inline test/unit/specs/core/edgeNetwork/mergeLifecycleResponses.spec.js:75]: A unit test that coves the "merge two arrays" code path would be good.

**Review by carterworks — APPROVED** [review]

File: src/core/edgeNetwork/mergeLifecycleResponses.js [hunk 2]
```diff
@@ -18,10 +19,25 @@ export default returnValues => {
...
   const lifecycleOnBeforeRequestReturnValues = returnValues;
 
-  return assign(
-    {},
+  return [
     ...lifecycleOnResponseReturnValues,
     ...consumerOnResponseReturnValues,
     ...lifecycleOnBeforeRequestReturnValues
-  );
+  ].reduce((accumulator, currentValue) => {
```
Comment (jasonwaters) [inline src/core/edgeNetwork/mergeLifecycleResponses.js:26]: This is the defect I was going to tell you about in our meeting the other day.  much improved 🏆

Comment (jasonwaters) [reply src/core/edgeNetwork/mergeLifecycleResponses.js:26]: i would really like this `accumulation` of results to be available as the lifecycle progresses.  I know we discussed it possibly being less performant, but might it be worth trying it out?  Changing from `promise.all` to a sequential evaluation where the responses are accumulated.  that still seems like an ideal solution for the decisioning engine propositions to be added.

Comment (jfkhoury) [inline src/core/edgeNetwork/mergeLifecycleResponses.js:26]: Would it help to extract this code into a separate util and unit test that alone?

**Review by jfkhoury — COMMENTED** [review]
Love the fix, and all the test updates!

**Review by carterworks — APPROVED** [review]

File: src/utils/assignConcatArrayValues.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import assign from "./assign";
```
Comment (jfkhoury) [inline src/utils/assignConcatArrayValues.js:12]: Look at that beauty :)

File: test/unit/specs/utils/assignConcatArrayValues.spec.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
+import assignConcatArrayValues from "../../../../src/utils/assignConcatArrayValues";
```
Comment (jfkhoury) [inline test/unit/specs/utils/assignConcatArrayValues.spec.js:1]: More beauty!!!

**Review by jfkhoury — APPROVED** [review]
Looks great @jonsnyder ! I think @jasonwaters 's comment is worth discussing, but it can be after the release.
